### ISSUE TITLE
Supporting relative paths for custom toolchain locations

### ISSUE THREAD

**Issue by vmadalin** [gradle/gradle#31659]
### Expected Behavior

Toolchain resolves relative path defined under `org.gradle.java.installations.paths` and using them to find a matching toolchain with the defined criteria

### Current Behavior (optional)

Right now when using relative paths for custom toolchain locations using `org.gradle.java.installations.paths` properties the following exception is thrown:
```
java.lang.UnsupportedOperationException: Cannot convert relative path ../jdk/jdk17/linux to an absolute file.
```

### Context

Considering custom environments this option might be interesting known that such paths might be added under project `gradle.properties` being visible to VCS. This will benefit task/projects toolchain but also for the daemon toolchain.

**Comment by ov7a** [comment]
This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

**Comment by big-guy** [comment]
I don't think we want to support this.  If anything, we should reduce the importance of `org.gradle.java.installations.paths` when using toolchains. 

I can discuss this more with Louis/Madalin.

**Comment by ljacomet** [comment]
Given this turns out to be supported for non daemon toolchains, adding support for daemon toolchains in 8.14 as we have report of use cases, that is projects where the JDK is available relative to the project root directory.

**Comment by ljacomet** [comment]
The attempted fix did not work, pushing this out of the 8.14 release.
